test/embedded/Inputs/tiny-runtime-dummy-refcounting.c

```diff
@@ -0,0 +1,51 @@
+#include <strings.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+size_t _swiftEmptyArrayStorage[] = { /*isa*/0, /*refcount*/0, /*count*/0, /*flags*/1 };
+
+typedef struct HeapObject {
+  void *metadata;
+  size_t refcount;
+} HeapObject;
+
+void *swift_allocObject(void *metadata, size_t requiredSize, size_t requiredAlignmentMask) {
+  void *r = NULL;
+  posix_memalign(&r, requiredAlignmentMask + 1, requiredSize);
+  bzero(r, requiredSize);
+  HeapObject *object = r;
+  object->metadata = metadata;
+  object->refcount = 1;
+  return object;
+}
+
+void swift_deallocClassInstance(HeapObject *object, size_t allocatedSize, size_t allocatedAlignMask) {
+  // don't deallocate ¯\_(ツ)_/¯
+}
+
+HeapObject *swift_initStackObject(void *metadata, HeapObject *object) {
+  object->metadata = metadata;
+  object->refcount = 1;
+  return object;
+}
+
+bool swift_isUniquelyReferenced_nonNull_native(HeapObject *object) {
+  return object->refcount == 1;
+}
+
+void swift_release(HeapObject *object) {
+  // don't release ¯\_(ツ)_/¯
```

Comment:
Could you still fix up the refcount here? I guess it doesn't matter because it's for a test, but then why retain it? Maybe there are some uniqueness things that are still important?

Reply:
Aha, yes, that's the answer. Array instantiation has a "isUniquelyReferenced" internal check, so right after creation of a new class, refcount must be 1, but it felt too wrong to me to have isUniquelyReferenced lie. Let me actually implement more logic here.

Reply:
How's it looking now?

Reply:
Looks great!